File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.go
func findThePrefixCommonArray(A []int, B []int) []int {
    n := len(A)
    ans := make([]int, n)

    presentA := make([]bool, n + 1)
    presentB := make([]bool, n + 1)
    cnt := 0

    for i := 0; i < n; i++ {
        if A[i] == B[i] {
            cnt++
        }
        if presentA[B[i]] {
            cnt++
        }
        if presentB[A[i]] {
            cnt++
        }

        presentA[A[i]] = true
        presentB[B[i]] = true

        ans[i] = cnt
    }
    return ans
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function(A, B) {
    let n = A.length;
    const ans = []
    const pa = Array(n + 1).fill(false)
    const pb = Array(n + 1).fill(false)
    var cnt = 0

    for (let i = 0; i < n; i++) {
        cnt += A[i] == B[i] ? 1 : 0;
        cnt += pa[B[i]] ? 1 : 0;
        cnt += pb[A[i]] ? 1 : 0;

        pa[A[i]] = true
        pb[B[i]] = true
        ans.push(cnt)
    }
    return ans
};